### PR TITLE
feat: Datepicker & Date-time picker

### PR DESCRIPTION
**Pull request by maulik-thaker1**
## Summary

Datepicker component with two variants : Datepicker and date with time picker

## ADO Story 

https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/1238186

## Checklist

- [x] Used Conventional Commit messages as outlined in the contributing guide.
  - [x] Noted breaking changes (if any).
- [x] Documented/updated all props, events, slots, parts, etc with JSDoc.
  - [x] Ran the `analyze` command to update Storybook docs.
- [x] Added/updated Stories with controls to cover all variants.
- [x] Added/updated a basic unit test, at least, for each component.
- [x] Added/updated the Figma link for the Story's Design tab.
- [x] Added any new component exports to the src/index.ts file

### CODE REVIEW

File: src/components/reusable/datePicker/datepicker.scss [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  right: 16px;
+  transform: translateY(-50%);
+  cursor: pointer;
+  background: var(--kd-color-background-inverse);
+  display: block;
+
+  kd-icon {
+    display: block;
+  }
+}
```
Comment (brian-patrick-3) [inline src/components/reusable/datePicker/datepicker.scss:127]: I don't think this is used?

**Review by brian-patrick-3 — CHANGES_REQUESTED** [review]
Seems pretty good! I did notice one thing that seems weird: when I type in the field it removes focus entirely, which doesn't seem right when I compare to a simple `<input type="date">`.

**Comment by brian-patrick-3** [comment]
> focus stays when type by keyboard or arrow up / down

Only after there's already an entry. The first time I try to type when it's empty, it loses focus.

File: src/components/reusable/datePicker/datepicker.scss (hunk 1 above)
Comment (maulik-thaker1) [reply src/components/reusable/datePicker/datepicker.scss]: done

**Comment by maulik-thaker1** [comment]
> > focus stays when type by keyboard or arrow up / down
> 
> Only after there's already an entry. The first time I try to type when it's empty, it loses focus.

yeah its weird don't know whats happen there. Can you help me out here. Think updating component on-input cause this issue might be.

**Comment by maulik-thaker1** [comment]
> > yeah its weird don't know whats happen there. Can you help me out here. Think updating component on-input cause this issue might be.
> 
> I don't have time to dig into it, but it might be related to setting inputEl.value on updated. Try taking that out.

sure I am looking into it.

**Comment by maulik-thaker1** [comment]
> > > yeah its weird don't know whats happen there. Can you help me out here. Think updating component on-input cause this issue might be.
> > 
> > 
> > I don't have time to dig into it, but it might be related to setting inputEl.value on updated. Try taking that out.
> 
> sure I am looking into it. why we need inputEl.value here. Can we remove for type="date"

**Comment by maulik-thaker1** [comment]
> > > yeah its weird don't know whats happen there. Can you help me out here. Think updating component on-input cause this issue might be.
> > 
> > 
> > I don't have time to dig into it, but it might be related to setting inputEl.value on updated. Try taking that out.
> 
> sure I am looking into it. but removing inputEl.value not worked.

I tried looking all possibilities but not sure why its losing focus. Need to deep digger may be something happened about LIT component update lifecycle but its not blocking anything.

**Review by brian-patrick-3 — APPROVED** [review]

**Pull request merged**